src/python/pants/option/global_options.py

```diff
@@ -993,6 +993,29 @@ class BootstrapOptions:
             """
         ),
     )
+    native_options_validation = EnumOption(
+        default=NativeOptionsValidation.WARNING,
+        help=softwrap(
+            """
+            Pants is switching its option parsing system from a legacy parser written in Python
+            to a new one written in Rust.
+
+            The results of parsing a given option by each system should be identical. However
+            during a transition period we will run both parsers and compare their results.
+            This option controls how to report discrepancies that arise.
+
+            ERROR: Discrepancies will cause Pants to exit.
+            WARNING: Discrepancies will be logged but Pants will continue.
+            IGNORE: A last resort to turn off this check entirely.
+
+            If you encounter discrepancies that are not easily resolvable, please reach out to
+            us on Slack: https://www.pantsbuild.org/community/getting-help.
+
+            A future version of Pants will remove the legacy parser, so it is imperative that
```

Comment:
Do you have thoughts yet on which future version we should be targeting for removal?

Reply:
I think in 2.23.x we switch the canonical version returned by `for_scope()` to be the native parser's version, and in 2.24.x we remove the legacy parser.

Reply:
Ah, cool. Maybe we could add the 2.23.x reference to the message, to add a sense of urgency for providing feedback?

Reply:
Done